### DISCUSSION TITLE
How to execute a rename symbol for an unopened file

### DISCUSSION THREAD

**Discussion by xueerli** [TypeFox/monaco-languageclient#539] [Q&A]
I recently encountered an issue whilst using the ‘rename symbol’ feature on Monaco Editor. The problem arises specifically when trying to rename symbols for files that have not been opened. In this scenario, an error returns stating 'bad edit - model not found’.

As I understand it, the issue stems from the fact that Monaco does not create a model for unopened files. As a result, it can’t apply an edit. This process seems to work fine for opened files as I am able to rename symbols without any issues.

**Comment by CGNonofr** [comment]
Can you provide a bit of context? How are you using the library? If you want it to be able to create models of unopened files, you'll need to register a filesystem provider
